Source/Core/Common/TitleDatabase.cpp

```diff
@@ -0,0 +1,72 @@
+// Copyright 2017 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include <fstream>
+#include <iostream>
+
+#include "Common/FileUtil.h"
+#include "Common/StringUtil.h"
+#include "Common/TitleDatabase.h"
```

Comment:
This should be placed as the first include in the file (e.g. see [Core.cpp](https://github.com/dolphin-emu/dolphin/blob/master/Source/Core/Core/Core.cpp#L5)) as it prevents any includes above it from satisfying any dependencies in the header indirectly if they're missing.